testing/openjdk12/APKBUILD

```diff
@@ -0,0 +1,307 @@
+# Contributor: Simon Frankenberger <simon-alpine@fraho.eu>
+# Maintainer: Simon Frankenberger <simon-alpine@fraho.eu>
+pkgname=openjdk12
+pkgver=12.0.2_p10
+_pkgver=${pkgver/_p/+}
+pkgrel=0
+pkgdesc="Oracle OpenJDK 12"
+url="https://hg.openjdk.java.net/jdk-updates/jdk12u"
+arch="all !x86 !armhf !armv7" # openjdk11 is not yet available on 32 bit arches
+license="GPL-2.0 with Classpath"
+makedepends="autoconf
+bash
+gawk
+grep
+make
+openjdk11-jdk
+zip
+alsa-lib-dev
+cups-dev
+elfutils-dev
+fontconfig-dev
+freetype-dev
+giflib-dev
+lcms2-dev
+libexecinfo-dev
+libffi-dev
+libjpeg-turbo-dev
+libx11-dev
+libxext-dev
+libxrandr-dev
+libxrender-dev
+libxt-dev
+libxtst-dev
+linux-headers
+zlib-dev"
+depends="$pkgname-jmods $pkgname-demos $pkgname-doc $pkgname-jdk" # for the virtual openjdk12 package
+subpackages="$pkgname-jmods:_jmods:noarch
+$pkgname-demos:_demos:noarch
+$pkgname-doc:_doc:noarch
+$pkgname-dbg:_dbg
+$pkgname-jre:_jre
+$pkgname-src:_src:noarch
+$pkgname-jre-headless:_jre_headless
+$pkgname-jdk:_jdk"
+source="jdk-$_pkgver.tar.bz2::http://hg.openjdk.java.net/jdk-updates/jdk12u/archive/jdk-$_pkgver.tar.bz2
+
+build.patch
+aarch64.patch
+arm.patch
+ppc64le.patch
+x86.patch
+
+HelloWorld.java
+TestECDSA.java
+TestCryptoLevel.java
+Alpine_Bug_10126.java
+"
+builddir="$srcdir/jdk12u-jdk-$_pkgver"
+
+_java_home="/usr/lib/jvm/java-12-openjdk"
+
+ldpath="$_java_home/lib:$_java_home/lib/server"
+sonameprefix="$pkgname:"
+
+# enable running the JTReg tests in check?
+# see comment in that function for explanation
+_run_jtreg=${_run_jtreg:-0}
+if [ $_run_jtreg -ne 0 ]; then
+	makedepends="$makedepends java-jtreg"
+	checkdepends="$checkdepends ttf-freefont xvfb-run"
+fi
+
+
+prepare() {
+	default_prepare
+
+	# update autoconf files to detect alpine
+	update_config_sub
+	update_config_guess
+
+	# remove not compilable module (hotspot jdk.hotspot.agent)
+	# this needs libthread_db which is only provided by glibc
+	#
+	# haven't found any way to disable this module so just remove it.
+	rm -r src/jdk.hotspot.agent
+}
+
+build() {
+	if [ $_run_jtreg -ne 0 ]; then
+		_with_jtreg="--with-jtreg=/usr/share/java/jtreg"
+	else
+		_with_jtreg="--with-jtreg=no"
+	fi
+
+	# CFLAGS, CXXFLAGS and LDFLAGS are ignored as shown by a warning
+	# in the output of ./configure unless used like such:
+	#  --with-extra-cflags="${CFLAGS}"
+	#  --with-extra-cxxflags="${CXXFLAGS}"
+	#  --with-extra-ldflags="${LDFLAGS}"
+	# See also paragraph "Configure Control Variables" from "common/doc/building.md"
+	# shellcheck disable=2097 disable=2098
+	CFLAGS='' CXXFLAGS='' LDFLAGS='' \
```

Comment:
Why is assigning empty values to these environment variables desirable?

Reply:
If these variables are set, the configuration prints ugly warnings. See the comment above